plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -362,7 +369,8 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
         BaseTable table = (BaseTable) loadTable(session, schemaTableName);
         validateTableCanBeDropped(table);
         try {
-            deleteTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+            Failsafe.with(CONCURRENT_MODIFICATION_EXCEPTION_RETRY_POLICY)
```

Comment:
I opted to apply retries specifically for DROP operations for concurrent modfication exception retry policy. Currently, the RENAME operations do not use retries.